pandas/_testing.py

```diff
@@ -2882,7 +2882,7 @@ def convert_rows_list_to_csv_str(rows_list: List[str]):
 
 def external_error_raised(
     expected_exception: Type[Exception],
-) -> Callable[[Type[Exception], None], None]:
+) -> ContextManager:
```

Comment:
instance of pytest.raises is a RaisesContext with is structurally a ContextManager, may want to refine the type here @WillAyd